src/cmd/common.go

```diff
@@ -215,3 +215,64 @@ func cliSetup(cmd *cobra.Command) error {
 
 	return nil
 }
+
+// Plan performs validation, confirmation and deployment of a bundle
+func plan(bndlClient *bundle.Bundle) error {
+	var err error
+	if bundleCfg.TofuOpts.IsTofu {
+		_, _, _, err = bndlClient.PreDeployValidationTF()
+	} else {
+		_, _, _, err = bndlClient.PreDeployValidation()
+	}
+	if err != nil {
+		return fmt.Errorf("failed to validate bundle: %s", err.Error())
+	}
+
+	// confirm deployment
+	if ok := bndlClient.ConfirmBundleDeploy(); !ok {
+		return fmt.Errorf("bundle deployment cancelled")
+	}
+
+	// deploy the bundle
+	if bundleCfg.TofuOpts.IsTofu {
+		// extract the tarballs of the Zarf Packages within the bundle
+		if err := bndlClient.Extract(bndlClient.GetDefaultExtractPath()); err != nil {
+			return fmt.Errorf("failed to extract packages from budnle: %s", err.Error())
+		}
+
+		// Determine the location of the local tfstate file
+		stateFilepath, err := filepath.Abs(bndlClient.GetTofuStateFilepath())
+		if err != nil {
+			return fmt.Errorf("failed to locate path to tfstate file: %s", err.Error())
+		}
+		stateFlag := fmt.Sprintf("-state=%s", stateFilepath)
+
+		// Determine the location of the extracted *.tf files
+		chdirFlag := fmt.Sprintf("-chdir=%s", filepath.Dir(bndlClient.GetDefaultExtractPath()))
+
+		// configure extra environments variables to use during future tofu commands
+		envMap := make(map[string]string)
+
+		// Configure tofu to use the terraformrc file that came with the bundle so it knows where to locate the providers
+		customRCPath := filepath.Join(filepath.Dir(bndlClient.GetDefaultExtractPath()), config.TerraformRC)
+		envMap["TF_CLI_CONFIG_FILE"] = customRCPath
+
+		// Custom envvar that the provider reads to know to use the extracted Zarf packages
+		envMap["UDS_LOCAL_PATH_OVERRIDE"] = bndlClient.GetDefaultExtractPath()
+
+		// Run the `tofu apply` command
+		os.Args = []string{"tofu", chdirFlag, "plan", "-input=false", stateFlag}
+		err = useEmbeddedTofu(envMap)
+		if err != nil {
+			message.Warnf("unable to deploy bundle that was built from a .tf file: %s", err.Error())
+			return err
+		}
+	} else {
+		if err := bndlClient.Deploy(); err != nil {
```

Comment:
case and point; this `else` block probably shouldn't exist, but I copied it ~2 minutes before giving the demo :D